calculator-app/src/App.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import KeyPad from './components/KeyPad'
+import './App.css';
+
+class App extends React.Component {
+  state = {
+    displayValue: "0",
+    previousValue: null,
+    operation:"",
+    waitingNewValue:false
+  }
+
+  handleNumberClear = (e) => {
+    this.setState({displayValue: 0, previousValue: null, operation:"", waitingNewValue:false})
```

Comment:
Having this 0 as a number instead of a string is breaking your decimal point. Since once you clear and then type `.` it tries to loop through a number. Which is not possible. Switch this to a string and it should be ok.